chain-signatures/node/src/protocol/signature.rs

```diff
@@ -603,13 +606,44 @@ impl SignatureManager {
                 failed_presigs.push(presignature);
                 continue;
             }
-            let presig_id = presignature.id;
+            let state_views = sig_participants
+                .iter()
+                .filter_map(|(p, _)| Some((*p, state_views.get(p)?)));
+
+            // Filter out the active participants with the state views that have the triples we want to use.
+            let stable_filtered = state_views
+                .filter(|(_, state_view)| {
```

Comment:
Again I think we should log parties with missing presignatures & warn if we have N - T presignatures missing